src/components/Comments.jsx

```diff
@@ -0,0 +1,50 @@
+import { fetchCommentsByArticleID } from "../api";
+import { useEffect, useState } from "react";
+import { sqlDateFormatter } from "../utils.js";
+
+const Comments = ({ article_id, comments, setComments }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (typeof article_id !== "undefined" && article_id !== "") {
```

Comment:
We shouldn't need this conditional logic here. The comments component should only be rendered from the SingleArticle component, so it should always have access to article_id. If that isn't the case, it's likely you were trying to render this component before the article fetch request was completed, but this should now be handled by your loading patterns.